util/grav_comp.hpp

```diff
@@ -41,6 +41,47 @@ class GravityCompensation
     Eigen::VectorXd comp_torque(num_dof);
     comp_torque.setZero();
 
+    // Wrench vector
+    Eigen::VectorXd wrench_vec(6); // For a single frame; this is (Fx/y/z, tau x/y/z)
+    wrench_vec.setZero();
+    for (size_t i = 0; i < num_frames; ++i)
+    {
+      // Set translational part
+      for (size_t j = 0; j < 3; ++j)
+      {
+        wrench_vec[j] = -normed_gravity[j] * masses[i];
+      }
+
+      // Add the torques for each joint to support the mass at this frame
+      comp_torque += jacobians[i].transpose() * wrench_vec;
+    }
+        return comp_torque;
+  }
+  static Eigen::VectorXd getEfforts(
```

Comment:
I think instead of duplicating the implementation, the implementation accepting a `GroupFeedback` can just call this implementation, right?

Reply:
We need to move these out of a class. This is writing .NET/Java code here, but we are using C++. Please just make these free standing (_i.e._ outside of a class definition). I understand that you were following the existing convention, though  @justin-h-m-tunis

Reply:
Yeah, this is my bad here -- sorry!